frontend/model/contracts/identity.js

```diff
@@ -40,6 +40,14 @@ DefineContract({
           Vue.delete(state.attributes, attribute)
         }
       }
+    },
+    'gi.contracts/identity/updateSettings': {
+      validate: object,
+      process ({ data }, { state }) {
+        for (var key in data) {
+          Vue.set(state.settings, key, data[key])
```

Comment:
`state.settings` will not exist, so this will throw an error. Make sure to define `state.settings` in the `process` function of `'gi.contracts/identity'`.

- Have a look at how the `process` function in `'gi.contracts/group'` does it using `merge` and `initialState` and copy that approach.
- Edit the `validate` on line 15 to use `objectMaybeOf` instead of `objectOf` (this will help with checkpointing (#354))

Reply:
Oh I did it before but removed it. Added back.